components/core/tests/test-ir_encoding_methods.cpp

```diff
@@ -0,0 +1,341 @@
+// Catch2
+#include "../submodules/Catch2/single_include/catch2/catch.hpp"
+
+// Project headers
+#include "../src/ffi/encoding_methods.hpp"
+#include "../src/ffi/ir_stream/encoding_methods.hpp"
+#include "../src/ffi/ir_stream/decoding_methods.hpp"
+#include "../src/ffi/ir_stream/protocol_constants.hpp"
+
+using ffi::decode_float_var;
+using ffi::decode_integer_var;
+using ffi::decode_message;
+using ffi::eight_byte_encoded_variable_t;
+using ffi::four_byte_encoded_variable_t;
+using ffi::encode_float_string;
+using ffi::encode_integer_string;
+using ffi::encode_message;
+using ffi::epoch_time_ms_t;
+using ffi::get_bounds_of_next_var;
+using ffi::VariablePlaceholder;
+using ffi::wildcard_query_matches_any_encoded_var;
+using ffi::ir_stream::get_encoding_type;
+using ffi::ir_stream::IRBuffer;
+using ffi::ir_stream::IRErrorCode;
+using ffi::ir_stream::TimestampInfo;
+using ffi::ir_stream::cProtocol::EightByteEncodingMagicNumber;
+using ffi::ir_stream::cProtocol::FourByteEncodingMagicNumber;
+using ffi::ir_stream::cProtocol::MagicNumberLength;
+using std::chrono::duration_cast;
+using std::chrono::milliseconds;
+using std::chrono::system_clock;
+using std::is_same_v;
+using std::string;
+using std::string_view;
+using std::vector;
+
+static epoch_time_ms_t get_current_ts () {
```

Comment:
Although this is a test file, it should still follow the standard of marking local methods as static, putting their prototypes at the top of the file, and adding a docstring if their name does not completely explain their functionality.